exporter/diagnostic_data_collector.go

```diff
@@ -43,15 +43,14 @@ func (d *diagnosticDataCollector) Collect(ch chan<- prometheus.Metric) {
 	res := d.client.Database("admin").RunCommand(context.TODO(), cmd)
 
 	if err := res.Decode(&m); err != nil {
-		ch <- prometheus.NewInvalidMetric(prometheus.NewInvalidDesc(err), err)
```

Comment:
Why can't we do both?
Log error and return invalid metric?

Reply:
If we do that, prometheus will stop receiveing metrics so, we are preventing all other metrics to be exposed

Reply:
got it